### ISSUE TITLE
Add a e2e for `ApplicationNotFound` scenario

### ISSUE THREAD

**Issue by DhritiShikhar** [redhat-developer/service-binding-operator#547]
Based on the conversation [here](https://github.com/redhat-developer/service-binding-operator/pull/498#issuecomment-654759801), an e2e should be added for `ApplicationNotFound` scenario.

**Comment by pmacik** [comment]
@DhritiShikhar Is the following scenario - written in the Gherkin language for the acceptance tests framework - accurate?

```gherkin
Feature: Bind an application to a service

    Background:
        Given Namespace [TEST_NAMESPACE] is used
        And Service Binding Operator is running
        And PostgreSQL DB operator is installed

    @negative
    Scenario: Service Binding Request without application selector
        Given Imported Nodejs application "nodejs-empty-app" is running
        And DB "db-demo-empty-app" is running
        When Service Binding Request is applied to connect the database and the application
            """
            apiVersion: apps.openshift.io/v1alpha1
            kind: ServiceBindingRequest
            metadata:
                name: binding-request-empty-app
            spec:
                backingServiceSelector:
                    group: postgresql.baiju.dev
                    version: v1alpha1
                    kind: Database
                    resourceRef: db-demo-empty-app
            """
        Then jq ".status.conditions[] | select(.type=="CollectionReady").status" of Service Binding Request "binding-request-empty-app" should be changed to "True"
        And jq ".status.conditions[] | select(.type=="InjectionReady").status" of Service Binding Request "binding-request-empty-app" should be changed to "False"
```

**Comment by DhritiShikhar** [comment]
@pmacik What you've mentioned here is `EmptyApplicationSelector` scenario where the user doesn't specify the application selector in SBR.

We definitely need acceptance test for this scenario.

In addition to that, we need acceptance test for `ApplicationNotFound` scenario which means that the user has specified the application in the SBR but that application cannot be found.

**Comment by pmacik** [comment]
@DhritiShikhar Of course :), I was just looking into the referenced #498 and got inspired there: We'll definitely use the above `EmptyApplicationSelector`, as well. So for the `ApplicationNotFound` scenario, is the following accurate (assuming the current master version of SBO)? :)

```gherkin

Feature: Bind an application to a service

    Background:
        Given Namespace [TEST_NAMESPACE] is used
        And Service Binding Operator is running
        And PostgreSQL DB operator is installed

... 

    @negative
    Scenario: Attempt to bind an non existing application to PostgreSQL database
        Given DB "db-demo-missing-app" is running
        And Imported Nodejs application "nodejs-missing-app" is not running
        When Service Binding Request is applied to connect the database and the application
            """
            apiVersion: apps.openshift.io/v1alpha1
            kind: ServiceBindingRequest
            metadata:
                name: binding-request-missing-app
            spec:
                applicationSelector:
                    resourceRef: nodejs-missing-app
                    group: apps
                    version: v1
                    resource: deployments
                backingServiceSelector:
                    group: postgresql.baiju.dev
                    version: v1alpha1
                    kind: Database
                    resourceRef: db-demo-missing-app
            """
        Then jq ".status.conditions[] | select(.type=="Ready").status" of Service Binding Request "binding-request-missing-app" should be changed to "False"
        And jq ".status.conditions[] | select(.type=="Ready").reason" of Service Binding Request "binding-request-missing-app" should be changed to "BindingFail"
        And jq ".status.conditions[] | select(.type=="Ready").message" of Service Binding Request "binding-request-missing-app" should be changed to ".* not found"
```

**Comment by pmacik** [comment]
@DhritiShikhar I've [included](https://github.com/redhat-developer/service-binding-operator/pull/541/files#diff-8600e044b29c6b69858c274f1fefbd3bR167-R218) (while temporarily disabled) those 2 scenarios in the acceptance tests PR (#541) so when that is merged, this issue would be resolved.
Could you please take a look? Thx

**Comment by DhritiShikhar** [comment]
@pmacik Yes. I will take a look

**Comment by pmacik** [comment]
@DhritiShikhar The `ApplicationNotFound` is implemented and merged as part of #541. Do you thing that this can be closed as done?

**Comment by jasperchui** [comment]
closing. Reopen if needed. Thanks.
